client/src/app/singleton.js

```diff
@@ -39,5 +39,5 @@ export function galaxyIsInitialized() {
 // Having a CORS issue in the toolshed iframe, store separate versions
 // of galaxy in each window for the short-term
 export function getStorage() {
-    return window;
+    return window.parent;
```

Comment:
@dannon this was necessary so that clicking on the rerun button from within the center iframe we don't nest the Galaxy app twice ... it seems this doesn't break anything else, but I also have no clue whether that's correct.

Reply:
I don't remember all the details right now, but I think this *probably* breaks what this was intended to fix by always returning a single toplevel Galaxy object.

Reply:
The source PR was this:  https://github.com/galaxyproject/galaxy/pull/7135, but I think that predates the shift to the new tool install interface, so we might be okay to actually revert all that and clean up/simplify a bit.